pkg/installmanager/installmanager.go

```diff
@@ -1874,6 +1882,35 @@ func isDirNonEmpty(dir string) bool {
 	return err == nil
 }
 
+// pasteInProviderCredentials Add the credentials from a given secret into the ic if nor present
+func pasteInProviderCredentials(icData []byte, cd *hivev1.ClusterDeployment) ([]byte, error) {
+	switch {
+	case cd.Spec.Platform.Nutanix != nil:
+		ic := installertypes.InstallConfig{}
+		if err := yaml.Unmarshal(icData, &ic); err != nil {
+			return nil, errors.Wrap(err, "could not unmarshal InstallConfig")
+		}
+
+		nutanixUsername := os.Getenv(constants.NutanixUsernameEnvVar)
+		nutanixPassword := os.Getenv(constants.NutanixPasswordEnvVar)
+		if nutanixUsername == "" || nutanixPassword == "" {
+			return nil, fmt.Errorf("invalid credentials for %s platform provider", constants.PlatformNutanix)
```

Comment:
Let's instead say
```suggestion
			return nil, fmt.Errorf("missing credentials for %s platform provider", constants.PlatformNutanix)
```

It's logically the difference between a 400 and a 403.